### PR TITLE
Update docs and make test cases more robust

### PR DESCRIPTION
**Pull request by Kshitij68**
# Description:

Update docstrings 

Make test cases for default_docs more robust

- [x] Changelog entry

### CODE REVIEW

File: kartothek/io/eager.py [hunk 1]
```diff
@@ -393,6 +387,18 @@ def commit_dataset(
         Input partition to be committed.
 
     """
+    if output_dataset_uuid is not None:
+        warnings.warn(
+            "The keyword `concat_partitions_on_primary_index` is deprecated and will be removed in the next major release.",
```
Comment (lr4d) [inline kartothek/io/eager.py]: wrong keyword `concat_partitions_on_primary_index`

File: kartothek/io/eager.py [hunk 2]
```diff
@@ -280,7 +274,7 @@ def read_table(
         >>> df = read_table(store, 'dataset_uuid', 'core')
 
     """
-    if concat_partitions_on_primary_index:
+    if concat_partitions_on_primary_index is not False:
```
Comment (lr4d) [inline kartothek/io/eager.py]: why was this change necessary?

File: kartothek/io/eager.py [hunk 3]
```diff
@@ -608,6 +614,24 @@ def write_single_partition(
     -------
     An empty :class:`~kartothek.io_components.metapartition.MetaPartition` referencing the new files
     """
+    if metadata is not None:
+        warnings.warn(
+            "The keyword `metadata` is deprecated and will be removed in the next major release.",
+            DeprecationWarning,
+        )
+
+    if overwrite is not False:
```
Comment (lr4d) [inline kartothek/io/eager.py]: is this deprecated?

File: kartothek/io/eager.py (hunk 1 above)
Comment (lr4d) [reply kartothek/io/eager.py]: is this deprecated?

File: kartothek/io/eager.py [hunk 4]
```diff
@@ -393,6 +387,18 @@ def commit_dataset(
         Input partition to be committed.
 
     """
+    if output_dataset_uuid is not None:
+        warnings.warn(
+            "The keyword `concat_partitions_on_primary_index` is deprecated and will be removed in the next major release.",
+            DeprecationWarning,
+        )
+
+    if df_serializer is not None:
```
Comment (lr4d) [inline kartothek/io/eager.py]: is this deprecated?

File: tests/core/test_docs.py [hunk 5]
```diff
@@ -76,9 +76,33 @@
...
+        store: str
+            This is an argument
+
+        """
+
+    resultant_docstrings = str(dummy_function.__doc__)
+    assert "This is a dummy_function" in resultant_docstrings
+    assert "This is an argument" in resultant_docstrings
+    assert (
+        "Factory dummy_function producing a KeyValueStore" not in resultant_docstrings
```
Comment (lr4d) [inline tests/core/test_docs.py]: Should be `Factory function producing a KeyValueStore.`. Also, can you add a short comment/docstring here to explain what you're doing?

File: kartothek/io/eager.py (hunk 1 above)
Comment (Kshitij68) [reply kartothek/io/eager.py]: The argument is not used anywhere. Made it clear in the warning statement also.

File: kartothek/io/eager.py (hunk 2 above)
Comment (Kshitij68) [reply kartothek/io/eager.py]: Passing `concat_partitions_on_primary_index` as anything should also raise a warning. 
If not making the above change, passing 0 will not raise the warning

File: kartothek/io/eager.py (hunk 4 above)
Comment (Kshitij68) [reply kartothek/io/eager.py]: The argument has no use. Made it clear in the warning itself.

File: kartothek/io/eager.py (hunk 1 above)
Comment (Kshitij68) [reply kartothek/io/eager.py]: The argument has no use and hence put the warning. Made the warning statement more clear

File: kartothek/io/eager.py (hunk 3 above)
Comment (Kshitij68) [reply kartothek/io/eager.py]: same reasoning as others

File: kartothek/io/iter.py [hunk 6]
```diff
@@ -200,21 +200,23 @@ def update_dataset_from_dataframes__iter(
...
     Parameters
     ----------
-    df_generator: Iterable[Union[pd.DataFrame, Dict[str, pd.DataFrame]]]
-        The dataframe(s) to be stored.
 
     Returns
     -------
     The dataset metadata object (:class:`~kartothek.core.dataset.DatasetMetadata`).
     """
+    if load_dynamic_metadata is True:
```
Comment (fjetter) [inline kartothek/io/iter.py]: should be `is not None` since the entire parameter should've been removed in `3.0`

File: kartothek/core/docs.py [hunk 7]
```diff
@@ -129,24 +124,45 @@
...
+        Dask bag partition size. Use a larger numbers to decrease scheduler load and overhead, use smaller numbers for a
+        fine-grained scheduling and better resilience against worker errors.""",
+    "metadata_storage_format": """
+    metadata_storage_format: str
+        Optional list of datastorage format to use. Currently supported is `.json` & `.msgpack.zstd"`""",
+    "df_generator": """
+    df_generator: Iterable[Union[pd.DataFrame, Dict[str, pd.DataFrame]]]
+        The dataframe(s) to be stored""",
+    "central_partition_metadata": """
+    central_partition_metadata: TODO
```
Comment (fjetter) [inline kartothek/core/docs.py]: this keyword should no longer exist (should've been removed in 3.0 already). In any case, it doesn't have any effect anymore. Can you find the usage and add also warnings, please?

File: kartothek/core/docs.py [hunk 8]
```diff
@@ -129,24 +124,45 @@
...
-    partition_size: int
-        Amount of metapartitions to cluster into one dask partition.
-""",
+    partition_size: Optional[int]
+        Dask bag partition size. Use a larger numbers to decrease scheduler load and overhead, use smaller numbers for a
+        fine-grained scheduling and better resilience against worker errors.""",
+    "metadata_storage_format": """
+    metadata_storage_format: str
+        Optional list of datastorage format to use. Currently supported is `.json` & `.msgpack.zstd"`""",
+    "df_generator": """
```
Comment (fjetter) [inline kartothek/core/docs.py]: Not sure if I would count this as a "general" doc string. We usually kept very specific parameter documentation with the function it refers to. Is this used at more than one places?

File: kartothek/io/iter.py (hunk 6 above)
Comment (Kshitij68) [reply kartothek/io/iter.py]: Aah, stupid mistake. Thanks.

I would recommend it being `is not True` as the default value is `True`

File: kartothek/core/docs.py (hunk 8 above)
Comment (Kshitij68) [reply kartothek/core/docs.py]: Yes, it is used in `store_dataframes_as_dataset__iter` and `update_dataset_from_dataframes__iter`

File: kartothek/core/docs.py (hunk 7 above)
Comment (Kshitij68) [reply kartothek/core/docs.py]: You are referring to `central_partition_metadata` right? This was used `update_dataset_from_dataframes__iter`. I have depreciated from that place.

**Comment by lr4d** [comment]
@Kshitij68 build seems to be failing due to formatting. You can run the pre-commit hooks to fix this
